### DISCUSSION TITLE
Real time stream on browser

### DISCUSSION THREAD

**Discussion by cothingsusr** [bluenviron/mediamtx#569] [Questions]
Thanks @aler9 for this interesting project. I have a general question: is there a way to display a stream into browser (Microsoft edge) with 0 latency from ffmpeg or Gstreamer source (the browser + ffmpeg/GStreamer can be hosted both on same machine)?
Right now using rtsp server + hls, we can display the stream in browser but this is giving a latency of at least 1-2 seconds (in my tests I got more than 2 seconds). 
Thanks a lot for your support.

**Comment by aler9** [comment]
Hello, available methods are:

* Low-latency HLS (not implemented yet here, but it will be)
* WebRTC (see here for bridging from RTSP to WebRTC https://github.com/aler9/rtsp-simple-server/issues/566#issuecomment-917183566)
* wasm-compiled H264 decoder inside the frontend. This is in my opinion the fastest one, but you have to discard RTSP and build a custom backend implementation. The frontend code is here https://github.com/mbebenita/Broadway

**Reply by cothingsusr** [reply]
Thanks @aler9 for your response

[a comment by an automated account is not shown]
